watch_service.go

```diff
@@ -276,56 +316,49 @@ func (ws *watchService) repoWatcher(projectName, repoName, pathPattern string) (
 
 func (w *Watcher) start() {
 	if atomic.CompareAndSwapInt32(&w.state, initial, started) {
-		w.scheduleWatch(0)
+		go w.scheduleWatch()
```

Comment:
Since `Watcher` start only called once. So each watcher has only one daemon to do watch file and notify listeners (which are running on others routines) about changes

Reply:
Yeah but, `watchSchedule()` calls `doWatch()` and `doWatch()` calls `doWatchFunc()`.
In `doWatchFunc()`, `watchFile` is called, and `watchRequest()` is called which spawns another Go routine.
Am I missing something? :)

Reply:
Yeah you are right. But it's blocking until `watchRequest()` done, then no more spawn.
In short, there are only one routine additionally. So it's fine :)

It's the problem of old design, so this patch only focus on Watcher. Client and watchService should be next PR. WDYT @minwoox?

Reply:
Alright~! Let's fix the problem of the old design in next PR. Thank you~!

Reply:
Thank you very much for hard review ^_^

Reply:
Just don't know much about Golang. 😄